src/implementation/Server/HTTPServer/pubsub.ts

```diff
@@ -46,12 +47,12 @@ export default class HTTPServerPubSub implements IServerPubSub {
         await cb(data);
       } catch (e) {
         this.logger.error(`subscribe failed: ${e}`);
-        return res.send({ success: false });
+        return res.send({ status: HttpResponseStatusForSubscription.RETRY });
```

Comment:
Can we add a test to ensure this does not keep occurring? If the subscribe keeps on failing, we would have to keep on retrying, which could trigger an overload. Maybe we need a back-off timer with a max count here? Looking for input from everyone here :) (@shubham1172)

Reply:
Something like an exponential back-off and retries till like some static/configurable number of times. Else, at end, passing it to DROP. But, won't that mean that we would have to track the traceID and keep a count that how many times we have re-tried it?

Reply:
Note: to clarify, this is when a message comes in and we want to process it locally in our application. The logic is to ensure we can correctly process the message. We should maybe rename the logging from `this.logger.error(`subscribe failed: ${e}`);` to `this.logger.error(`[route-${topic}]  Could not process the incoming message, retrying`);`

I think it could be done simpler, `this.server.getServer().post` will create a route on the HTTP Server, so we can in the `HTTPServerPubSub` create an object that counts for the specific route how many times it was retried and the corresponding backoff timer. Whenever we succeed we then reset this timer and counter.

Reply:
> If the subscribe keeps on failing, we would have to keep on retrying, which could trigger an overload. Maybe we need a back-off timer with a max count here?


@XavierGeerinck, failing to ACK a message successfully will result in Dapr failing to ACK the message to the broker. Since the brokers already have backoffs, max retries, etc. in place, does the app need to worry about it? This also means making these options configurable and potential clashes with broker configuration. Regarding backoff, the consumer can only respond back with HTTP response with a delay, this would require us to hold HTTP connections.

Here is an example of these configurations in Redis - https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-redis-pubsub/ 

> We should maybe rename the logging from this.logger.error(subscribe failed: ${e}); to this.logger.error([route-${topic}] Could not process the incoming message, retrying);

Agreed! Maybe "attempting retry" since the broker may or may not actually respond back.